xarray/core/variable.py

```diff
@@ -1150,7 +1145,7 @@ def to_numpy(self) -> np.ndarray:
         data = self.data
 
         # TODO first attempt to call .to_numpy() once some libraries implement it
-        if isinstance(data, dask_array_type):
+        if hasattr(data, "chunks"):
             data = data.compute()
```

Comment:
`data.compute()` should load our lazy indexing classes too so I *think* we could safely call it without any checks. Then we could get rid of the `np.asarray` below

Reply:
Not sure what you mean - calling `data.compute()` with no checks just raises for every type of array except dask:

```
FAILED xarray/tests/test_dataarray.py::TestDataArray::test_to_masked_array - AttributeError: 'numpy.ndarray' object has no attribute 'compute'
FAILED xarray/tests/test_dataarray.py::TestNumpyCoercion::test_from_numpy - AttributeError: 'numpy.ndarray' object has no attribute 'compute'
FAILED xarray/tests/test_dataarray.py::TestNumpyCoercion::test_from_dask - AttributeError: 'numpy.ndarray' object has no attribute 'compute'
FAILED xarray/tests/test_dataarray.py::TestNumpyCoercion::test_from_pint - AttributeError: Method compute only implemented for objects of <class 'dask.array.core.Array'>, not <class 'nump...
FAILED xarray/tests/test_dataarray.py::TestNumpyCoercion::test_from_sparse - AttributeError: 'COO' object has no attribute 'compute'
FAILED xarray/tests/test_dataarray.py::TestNumpyCoercion::test_from_cupy - AttributeError: 'ndarray' object has no attribute 'compute'
```

Reply:
oh sorry. my bad. I thought we had xarray objects at this point.

Reply:
In general this conversion to numpy is the least-standardized part left as far as I can see.

It would be nice if there was a `to_numpy` method made part of the array API standard...